snap/validate_test.go

```diff
@@ -225,6 +227,9 @@ version: 1.0
 }
 
 func (s *ValidateSuite) TestIllegalHookName(c *C) {
```

Comment:
This is a strange test. We explicitly tell the hook is valid by faking a valid hook pattern that accepts everything, then complain that it's not valid?